ci: update components repo unit tests job commit [patch version]

devversion opened the pull request:
Patch version of #37176. Since #37176 now updates to TS 3.9 and the patch branch does not have support for that, this PR differs. It updates to a commit that supposedly fixes the flaky components-repo-unit-tests job failure.

devversion commented:
Note: looks like CI failure is unrelated. See https://github.com/angular/angular/pull/37151.

josephperrott reviewed (approved):
LGTM, looks like the CI failures are being corrected in another PR.

devversion committed:
ci: update components repo unit tests job commit

Updates the commit the `components-repo-unit-tests` job runs
against. The goal is that we run against a revision that at
least contains: https://github.com/angular/components/pull/19273.

The new commit contains fixes for a flaky test in the datepicker that we
saw failing in the components-repo-unit-tests job too:
https://app.circleci.com/pipelines/github/angular/angular/15359/workflows/27ffae7c-a7b8-46a3-9c9e-6dd22ca4734d/jobs/712643.

kara commented:
Closed in https://github.com/angular/angular/commit/784cf46a93168099d21680d18ddcab82e22e8d50
